packages/material-ui-styles/src/createBox.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { chainPropTypes } from '@material-ui/utils';
+import withStyles from './withStyles';
+
+function omit(input, fields) {
+  const output = {};
+
+  Object.keys(input).forEach(prop => {
+    if (fields.indexOf(prop) === -1) {
+      output[prop] = input[prop];
+    }
+  });
+
+  return output;
+}
+
+const withPropsStyles = style => {
+  const withPropsStyles2 = Component => {
+    return props => {
+      const proxy = theme => style(theme, props);
+      const component = withStyles(proxy)(Component);
```

Comment:
So `withPropsStyles` is a higher-order component that creates a component that creates a new component on every render? I guess that is ok but I would prefer a simpler solution that doesn't require re-mounting on every render. At least I think that's what happening which would result in re-mounting of the children, no?

Reply:
It also has an implication with CSS specificity. You can hardly override a Box without !important.

Reply:
Fixed.